licensing/src/main/java/io/crate/license/LicenseService.java

```diff
@@ -97,7 +96,11 @@ public void registerLicense(final LicenseKey licenseKey,
 
     static DecryptedLicenseData licenseData(DecodedLicense decodedLicense) throws IOException {
         byte[] decryptedContent = decryptLicenseContent(decodedLicense.type(), decodedLicense.encryptedContent());
-        return DecryptedLicenseData.fromFormattedLicenseData(decryptedContent);
+        return DecryptedLicenseData.fromFormattedLicenseData(decryptedContent,
+            () -> (decodedLicense.type() == LicenseType.TRIAL)
```

Comment:
There is no need for a `IntSupplier` here - would be possible to pass in the number directly.